### PR TITLE
Prac 07 feedback

### PR DESCRIPTION
**Pull request by Winminoo4444**
Please review my code @Nex-opensourcehorray

### CODE REVIEW

**Comment by Nex-opensourcehorray** [comment]
Brother that kind of fast le...

File: prac_07/guitar.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    def get_age(self):
+        """Function for getting the age of guitar"""
+        return CURRENT_YEAR - self.year
+
+    def is_vintage(self):
+        """Function to determine whether the guitar is vintage or not """
+        return self.get_age() >= VINTAGE_YEAR
+
+    def __lt__(self, other):
```
Comment (Nex-opensourcehorray) [inline prac_07/guitar.py:24]: Overall looks ok lah.

File: prac_07/language_file_reader.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+"""
+CP1404/CP5632 Practical
+File and class example - opens/reads a file, stores in objects of custom class
+(contains multiple versions for demonstration: using csv and namedtuple)
+"""
+
```
Comment (Nex-opensourcehorray) [inline prac_07/language_file_reader.py:6]: we see the pointer arithmetic attitude is added, quite good.

File: prac_07/programming_language.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        self.year = year
+
+    def __repr__(self):
+        """Return string representation of a ProgrammingLanguage."""
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, First appeared in {self.year}"
+
+    def is_dynamic(self):
+        """Determine if language is dynamically typed."""
+        return self.typing == "Dynamic"
+
```
Comment (Nex-opensourcehorray) [inline prac_07/programming_language.py:21]: This is a slightly small problem. The requirement is to add another method to identify the pointer arithmetic, so it is the same for the run test function.

File: prac_07/languages.csv [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Language,Typing,Reflection,Year
```
Comment (Nex-opensourcehorray) [inline prac_07/languages.csv:1]: Your other column for the pointer arithmetic and the datasets?
and I also believe that the requirement for you is to add another language piece of information to the file.

File: prac_07/myguitars.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+            name, year, cost = line.strip().split(',')
+            year = int(year)
+            cost = float(cost)
+            guitar = Guitar(name, year, cost)
+            guitars.append(guitar)
+    sorted_guitars = sorted(guitars)
+    for guitar in sorted_guitars:
+        print(guitar)
+
+
```
Comment (Nex-opensourcehorray) [inline prac_07/myguitars.py:18]: where are your get user input function, display function with showing vintage or not, and also your write back to CSV function?

File: prac_07/project.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        """Function for returning string representation"""
+        return f"{self.name}, start: {self.start_date.strftime('%d/%m/%Y')}, priority {self.priority}, " \
+               f"estimate: ${self.cost_estimate:.2f}, completion: {self.completion_percentage}% "
+
+    def __repr__(self):
+        """Funciton for returning string representation"""
+        return f"{self.name}, start: {self.start_date.strftime('%d/%m/%Y')}, priority {self.priority}," \
+               f" estimate: ${self.cost_estimate}, completion: {self.completion_percentage}% "
+
+    def __lt__(self, other):
```
Comment (Nex-opensourcehorray) [inline prac_07/project.py:20]: Did your is_completed method leave away from your project.py sweet home?

**Comment by Nex-opensourcehorray** [comment]
I didn't feel delighted with this code review...

File: prac_07/languages.csv [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+Java,Static,Yes,1995
+C++,Static,No,1983
+Python,Dynamic,Yes,1991
+Visual Basic,Static,No,1991
+Ruby,Dynamic,Yes,1995
```
Comment (Nex-opensourcehorray) [inline prac_07/languages.csv:6]: Oh, I see, You change the language with class but have not even put in an extra programming language to file, as well as the Pointer Arithmetic dataset.

File: prac_07/language_file_reader.py [hunk 8]
```diff
@@ -13,28 +13,36 @@
 def main():
     """Read file of programming language details, save as objects, display."""
```
Comment (Nex-opensourcehorray) [inline prac_07/language_file_reader.py:14]: By far no problem la, cause this is just an example.

File: prac_07/programming_language.py [hunk 9]
```diff
@@ -39,4 +40,4 @@ def run_tests():
```
Comment (Nex-opensourcehorray) [inline prac_07/programming_language.py:36]: The requirement is as long as you add your pointer arithmetic value at the string and run the test function, it is just ok.

File: prac_07/myguitars.py [hunk 10]
```diff
@@ -15,4 +15,5 @@ def main():
         print(guitar)
```
Comment (Nex-opensourcehorray) [inline prac_07/myguitars.py:17]: Your file lost quite a lot of functions, like asking the user to input their guitar information, your display function with a lack of vintage identification, and also writing back to the CSV function.

File: prac_07/project.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+        """Function to sort the list of projects based on priority number."""
+        return self.priority < other.priority
```
Comment (Nex-opensourcehorray) [inline prac_07/project.py:22]: Where is your is_complete method? using to split between incomplete and completed projects.

File: prac_07/project_management.py [hunk 12]
```diff
@@ -11,6 +11,7 @@
 class_projects = []
 project_dictionary = {}
 
+
```
Comment (Nex-opensourcehorray) [inline prac_07/project_management.py:14]: Overall, it is acceptable, that you did the sorting and separating the completed and incomplete project, so make sure the method is in your own definition class.

**Comment by Winminoo4444** [comment]
Thank you for your review. Appreciate for the details and giving me opportunity to learn a lot from your review.

**Pull request merged**